client/src/components/events-components/EventCard.tsx

```diff
@@ -1,35 +1,34 @@
-import { Card, CardContent, CardDescription } from "../ui/card";
+import { useState } from "react";
+import { Card, CardContent, CardDescription, CardFooter } from "../ui/card";
+import { Button } from "../ui/button";
+import EventCreation from "./EventCreation";
+import type { EventType } from "@/utils/interfaces";
 
 interface EventProps {
-  event: Event;
+  event: EventType;
+  owns: boolean;
 }
 
-interface Event {
-  created_at: string;
-  description: string;
-  eventImage: string;
-  eventTime: string;
-  location: string;
-  latitude: number;
-  longitude: number;
-  organizerId: number;
-  id: number;
-  sport: string;
-  updated_at: string;
-}
-
-const EventCard = ({ event }: EventProps) => {
+const EventCard = ({ event, owns }: EventProps) => {
+  const [edit, setEdit] = useState(false);
```

Comment:
```suggestion
  const [isEditing, setIsEditing] = useState(false);
```
would `isEditing` be a better name?